### DISCUSSION TITLE
Trackpad Workspace Switching Improvements

### DISCUSSION THREAD

**Discussion by hejchristiandev** [zen-browser/desktop#2821] [Ideas]
Love the new 1.0.1-a.18 update and to see trackpad switching is a nice touch (pun wasn't intended but let's keep it) 🥳  
There are two improvements I would love to see;

1. Currently the two finger gesture swipe feels to be the wrong way around? If I swipe to the right, it switches into the Workspace on the left. If this is desired behavior then the option to reverse this would be appreciated.

2. I can only swipe to switch workspaces in areas that don't have tabs. This is fine if the number of tabs is small, but when the tab menu is full this leaves very little space to swipe, and I have to find it first. It would be better if we could two-finger swipe left/right anywhere in the sidebar. (This would better match the experience for many of us that are coming from Arc)

3. Currently there is no motion to the swipe/switch, which can feel a bit jarring or sudden. This can serve as a visual feedback that a workspace switch is about to happen, and also a delay. For example, if I start to swipe in Arc, it won't actually switch into the new workspace unless I've gone a certain distance and let go. This could prevent accidental switching too, while still allowing fluid vertical scrolling of the tabs in the tab menu.

https://github.com/user-attachments/assets/16dab083-034a-4126-9b40-b611f9d1fb60

I've recorded a little clip from Arc to better demonstrate what I mean by point 3, I hope it helps. Maybe one way to describe it is that there is resistance to the trackpad switching?

**Comment by kathytu** [comment]
How are you activating this feature? I've been trying to swipe left and right on the sidebar and nothing happens when I do. I really enjoy this feature in Arc.

**Reply by hejchristiandev** [reply]
It's a bit finicky at the moment; You need to swipe on an area with no normal tabs. For me it works to swipe over the workspace icons at the bottom, or over pinned / essential tabs :)

**Reply by kathytu** [reply]
I think maybe this is still an issue (re: https://github.com/zen-browser/desktop/issues/4084#issue-2763703949)

**Reply by hejchristiandev** [reply]
> I think maybe this is still an issue (re: [#4084 (comment)](https://github.com/zen-browser/desktop/issues/4084#issue-2763703949))

Ah interesting - didn't realize it was broken completely for some folks - sorry that's been happening to you.

**Reply by kathytu** [reply]
Thanks for responding! It led me to the other issue, will keep an eye on it.

**Reply by hejchristiandev** [reply]
Just a headsup that this is now working across the entire sidebar as of https://zen-browser.app/release-notes/#1.7.4b 🥳 So hopefully that'll work for you too with the update.

I believe I saw somewhere the main dev has a macbook now? So hopefully those trackpad gestures and movements will get even better now. 👌

Still some things I'd love to see implemented to really make it feel polished like it was in Arc, but this is a tremendous update that will actually see me switching between my 4 workspaces this way again 😅

**Comment by iandoesallthethings** [comment]
The swiping here still feels backwards. Multi finger swiping on macOS operates like moving the thing under the pointer as if you're dragging a piece of paper, so swiping to the right to move to the space to the right breaks that pattern.
